index.js

```diff
@@ -91,6 +91,20 @@ function findFocusElement(details, dialog): ?HTMLElement {
   }
 }
 
+function toggleDetails(details, open) {
+  // Don't update unless state is changing
+  if (open === details.hasAttribute('open')) return
+
+  const summary = details.querySelector('summary')
+  if (summary) {
+    // Toggle via clicking summary so it can be canceled by listeners wanting
+    // to prevent the toggle
+    summary.click()
+  } else {
```

Comment:
It'd be good to have a test with no `<summary>` as well.

Reply:
👍 Added a describe with no summary element that tests toggle open, close button clicking, and close on escape.